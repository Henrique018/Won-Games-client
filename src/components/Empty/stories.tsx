import { Story, Meta } from '@storybook/react/types-6-0';

import Empty, { EmptyProps } from '.';

export default {
	title: 'Empty',
	parameters: {
		backgrounds: {
			default: 'won-dark',
		},
	},
	args: {
		title: 'Your wishlist is empty',
		description: 'Nothing to see here',
		hasLink: true,
	},
	component: Empty,
} as Meta;

export const Basic: Story<EmptyProps> = (args) => <Empty {...args} />;
