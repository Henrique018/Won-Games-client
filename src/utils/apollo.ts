import { useMemo } from 'react';
import {
	ApolloClient,
	InMemoryCache,
	HttpLink,
	NormalizedCacheObject,
} from '@apollo/client';

let apolloClient: ApolloClient<NormalizedCacheObject>;

function createApolloClient() {
	return new ApolloClient({
		ssrMode: typeof window === 'undefined', //true
		link: new HttpLink({ uri: 'http://localhost:1337/graphql' }),
		cache: new InMemoryCache(),
	});
}

export function initializeApollo(initialState = {}) {
	//verify if an apolloClient instance exists
	const apolloClientGlobal = apolloClient ?? createApolloClient();

	if (initialState) {
		apolloClientGlobal.cache.restore(initialState);
	}

	//always initialize with SSR mode and empty cache
	if (typeof window === 'undefined') return apolloClientGlobal;
	apolloClient = apolloClient ?? apolloClientGlobal;

	return apolloClient;
}

export function useApollo(initialState = {}) {
	const store = useMemo(() => initializeApollo(initialState), [initialState]);
	return store;
}
