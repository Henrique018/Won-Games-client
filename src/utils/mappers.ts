import { QueryGames_games } from 'graphql/generated/QueryGames';
import {
	QueryHome_banners,
	QueryHome_sections_popularGames_highlight,
	QueryHome_sections_upcomingGames_highlight,
	QueryHome_sections_freeGames_highlight,
} from 'graphql/generated/QueryHome';

export const bannerMapper = (banners: QueryHome_banners[] | null) => {
	return (
		banners &&
		banners.map((banner) => ({
			img: `http://localhost:1337${banner.image?.url}`,
			title: banner.title,
			subtitle: banner.subtitle,
			buttonLabel: banner.button?.label,
			buttonLink: banner.button?.link,
			...(banner.ribbon && {
				ribbonText: banner.ribbon?.text,
				ribbonColor: banner.ribbon?.color,
				ribbonSize: banner.ribbon?.size,
			}),
		}))
	);
};

export const highlightMapper = (
	highlight:
		| QueryHome_sections_popularGames_highlight
		| QueryHome_sections_upcomingGames_highlight
		| QueryHome_sections_freeGames_highlight
		| null
		| undefined
) => {
	return (
		highlight && {
			title: highlight?.title,
			subtitle: highlight.subtitle,
			buttonLabel: highlight.buttonLabel,
			buttonLink: highlight.buttonLink,
			alignment: highlight.alignment,
			floatImage: `http://localhost:1337${highlight.floatImage?.url}`,
			backgroundImage: `http://localhost:1337${highlight.background?.url}`,
		}
	);
};

export const gamesMapper = (
	gamesArray: QueryGames_games[] | null | undefined
) => {
	return (
		gamesArray &&
		gamesArray.map((game) => ({
			title: game.name,
			img: `http://localhost:1337${game.cover?.url}`,
			price: game.price,
			slug: game.slug,
			developer: game.developers[0].name,
		}))
	);
};
