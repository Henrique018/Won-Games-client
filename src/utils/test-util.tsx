import { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import {
	cartContextDefaultValues,
	cartContextTypes,
	CartContext,
} from 'hooks/use-cart';

import { ThemeProvider } from 'styled-components';
import theme from 'styles/theme';

type CustomRenderProps = {
	cartProviderProps?: cartContextTypes;
} & Omit<RenderOptions, 'queries'>;

const customRender = (
	ui: ReactElement,
	{
		cartProviderProps = cartContextDefaultValues,
		...renderOptions
	}: CustomRenderProps = {}
) => {
	return render(
		<ThemeProvider theme={theme}>
			<CartContext.Provider value={cartProviderProps}>
				{ui}
			</CartContext.Provider>
		</ThemeProvider>,
		renderOptions
	);
};

export * from '@testing-library/react';
export { customRender as render };
