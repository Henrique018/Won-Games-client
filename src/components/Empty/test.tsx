import { render, screen } from 'utils/test-util';

import Empty from '.';

const props = {
	title: 'A simple title',
	description: 'a simple description',
};

describe('<Empty />', () => {
	it('should render correctly', () => {
		const { container } = render(<Empty {...props} hasLink />);

		expect(
			screen.getByRole('image', { name: /a gamer playing videogame/i })
		).toBeInTheDocument();

		expect(
			screen.getByRole('heading', { name: /a simple title/i })
		).toBeInTheDocument();

		expect(screen.getByText(/a simple description/i)).toBeInTheDocument();

		expect(
			screen.getByRole('link', { name: /go back to store/i })
		).toHaveAttribute('href', '/');

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should not render a link when hasLink is not passed', () => {
		render(<Empty {...props} />);

		expect(
			screen.queryByRole('link', { name: /go back to store/i })
		).not.toBeInTheDocument();
	});
});
