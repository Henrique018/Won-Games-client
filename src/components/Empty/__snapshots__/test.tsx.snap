// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render correctly 1`] = `
.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FAFAFA;
  cursor: pointer;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: linear-gradient( 180deg,#ff5f5f -14.51%,#f062c0 102.86%,#f23131 102.86% );
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c4:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 100%;
}

.c2 {
  color: #F231A5;
  font-size: 2.8rem;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
  text-align: center;
}

<main
  class="c0"
>
  <img
    alt="a gamer playing videogame"
    class="c1"
    role="image"
    src="/img/empty.svg"
  />
  <h2
    class="c2"
  >
    A simple title
  </h2>
  <p
    class="c3"
  >
    a simple description
  </p>
  <a
    class="c4"
    href="/"
  >
    <span>
      Go back to store
    </span>
  </a>
</main>
`;
